### PR TITLE
Slurm 21.08 Upgrade in 2.11

### PR DESCRIPTION
**Pull request by Nbot07A**
### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
Upgrading slurm to 21.08.8-2 
Emulating https://github.com/aws/aws-parallelcluster-cookbook/pull/1193#issue-1041784175
* Link to impacted open issues.
N/A
### Tests
* Describe the automated and/or manual tests executed to validate the patch.
N/A
* Describe the added/modified tests.
N/A

### References
* Link to related PRs in other packages (i.e. cookbook, node).
These PRs are also part of the upgrade:
https://github.com/aws/aws-parallelcluster-node/pull/431
https://github.com/aws/aws-parallelcluster/pull/4272
* Link to documentation useful to understand the changes.
https://www.schedmd.com/news.php
https://github.com/aws/aws-parallelcluster-cookbook/pull/1193
https://github.com/aws/aws-parallelcluster-node/pull/350/files
https://github.com/aws/aws-parallelcluster/pull/3452
 https://github.com/aws/aws-parallelcluster-cookbook/pull/1430
 https://github.com/aws/aws-parallelcluster-node/pull/302/commits/cc3bad5ebb303c93608878676a0ae6a6563f6662#diff-37d18a99e1edd3058be1192c765dcff454f9891b65cc21622200d59008e58169
https://github.com/aws/aws-parallelcluster-node/blob/develop/src/slurm_plugin/slurm_resources.py#L77
https://github.com/aws/aws-parallelcluster-node/blob/release-2.11/src/common/schedulers/slurm_commands.py#L49

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@ aws-parallelcluster-cookbook CHANGELOG
 
 This file is used to list changes made in each version of the AWS ParallelCluster cookbook.
 
+2.11.8
+-----
+
+**CHANGES**
+- Upgrade Slurm to version 21.08.8-2
```
Comment (hanwen-pcluste) [inline CHANGELOG.md]: Missing final period

**Comment by hanwen-pcluste** [comment]
Great work!

Please take a look at the PR description of https://github.com/aws/aws-parallelcluster-cookbook/pull/1193 and copy them to the `Description of changes` of this PR description.

File: recipes/slurm_install.rb [hunk 2]
```diff
@@ -25,6 +25,18 @@
   action :purge
 end
 
+slurm_build_deps = value_for_platform(
+  'ubuntu' => {
+    'default' => %w(libjson-c-dev libhttp-parser-dev),
+  },
+  'default' => %w(json-c-devel http-parser-devel)
```
Comment (hanwen-pcluste) [inline recipes/slurm_install.rb]: Please fix the rubocop check by changing 
`%w(libjson-c-dev libhttp-parser-dev),`
to
`['libjson-c-dev', 'libhttp-parser-dev']`
etc

**Review by demartinofra — COMMENTED** [review]
* Please do merge into a separate branch and remember to review commit messages before merging
* We need to align the contents of these files:
  * https://github.com/aws/aws-parallelcluster-cookbook/tree/develop/cookbooks/aws-parallelcluster-slurm/templates/default/slurm/head_node
  * https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/cookbooks/aws-parallelcluster-slurm/templates/default/slurm/compute/slurmd.service.erb
* Backport https://github.com/aws/aws-parallelcluster-cookbook/pull/1463

**Review by chenwany — COMMENTED** [review]
Do we also want to build slurm with jwt support with the update?

File: CHANGELOG.md (hunk 1 above)
Comment (chenwany) [inline CHANGELOG.md]: nit: Upgrade Slurm to version 21.08.

**Review by chenwany — APPROVED** [review]

**Pull request merged**